crates/sui-core/src/consensus_handler.rs

```diff
@@ -382,6 +382,21 @@ impl<
                                 .try_aggregate_signed_signature_mpc_output(*output.clone())
                                 .is_ok()
                         {
+                            let output = match &output.value {
+                                SignatureMPCOutputValue::Sign { sigs, .. } => {
+                                    let mut copied_output = output.clone();
+                                    copied_output.value = SignatureMPCOutputValue::Sign {
+                                        sigs: sigs.clone(),
+                                        aggregator_public_key: output
+                                            .auth_sig()
+                                            .authority
+                                            .0
+                                            .to_vec(),
+                                    };
+                                    &copied_output.clone()
+                                }
+                                _ => output,
+                            };
```

Comment:
Let me sure I understand what's going on here.
This code is being called everytime some validator decides to submit a "mpc output" transaction?
And since this is just one validator that submits, you can get his PK?
And every such submission is of a vector of signatures? is it for a single or multiple sessions?

Reply:
Is there any assumption here that the same aggregator is used for multiple sessions?

Reply:
Can any validator submit such output or only the one for the session?

Reply:
it's one session. Right now any validator, I'll add some verification that only the selected aggregator can.